extensions/ql-vscode/src/queryserver-client.ts

```diff
@@ -84,10 +87,27 @@ export class QueryServerClient extends DisposableObject {
     this.nextProgress = 0;
     this.progressCallbacks = {};
     this.evaluationResultCallbacks = {};
+
+  }
+
+  initLogger() {
+    let storagePath = this.opts.contextStoragePath;
+    let isCustomLogDirectory = false;
+    if (this.config.customLogDirectory) {
+      if (fs.existsSync(this.config.customLogDirectory) && fs.statSync(this.config.customLogDirectory).isDirectory()) {
+        storagePath = this.config.customLogDirectory;
+        isCustomLogDirectory = true;
+      } else if (this.config.customLogDirectory) {
+        helpers.showAndLogErrorMessage(`${this.config.customLogDirectory} is not a valid directory. Logs will be stored in a temporary workspace directory instead.`);
+      }
```

Comment:
If the custom path doesn't exist, maybe you can create it (with an info message). I think it's only an error if it already exists, but isn't a file.

Reply:
I made an attempt in https://github.com/github/vscode-codeql/pull/863/commits/708baaf1d8ef3f2e82e87dc5e961e34c4596139a, which _seems_ to do the right thing, but any tips on syntax/fixes are very welcome 😄